src/datasources/accounts/accounts.datasource.spec.ts

```diff
@@ -103,53 +175,180 @@ describe('AccountsDatasource tests', () => {
+        new CacheDir(`account_${address}`, ''),
+      );
+      expect(cached).toBeUndefined();
+
+      expect(mockLoggingService.debug).toHaveBeenCalledTimes(2);
+      expect(mockLoggingService.debug).toHaveBeenNthCalledWith(1, {
+        type: 'cache_hit',
+        key: `account_${address}`,
+        field: '',
+      });
+      expect(mockLoggingService.debug).toHaveBeenNthCalledWith(2, {
+        type: 'cache_miss',
+        key: `account_${address}`,
+        field: '',
+      });
+    });
   });
 
   describe('getDataTypes', () => {
-    it('returns data types successfully', async () => {
-      const dataTypeNames = [
-        faker.lorem.slug(),
-        faker.lorem.slug(),
-        faker.lorem.slug(),
+    it('returns all data types from the database successfully', async () => {
+      const dataTypes = [
+        { name: faker.lorem.slug() },
+        { name: faker.lorem.slug() },
+        { name: faker.lorem.slug() },
       ];
       await sql`
-        INSERT INTO account_data_types (name) VALUES
-        (${dataTypeNames[0]}),
-        (${dataTypeNames[1]}),
-        (${dataTypeNames[2]})
-      `;
+        INSERT INTO account_data_types ${sql(dataTypes, 'name')}`;
 
       const result = await target.getDataTypes();
 
       expect(result).toStrictEqual(
-        expect.arrayContaining([
-          {
-            id: expect.any(Number),
-            name: dataTypeNames[0],
-            description: null,
-            is_active: true,
-            created_at: expect.any(Date),
-            updated_at: expect.any(Date),
-          },
-          {
-            id: expect.any(Number),
-            name: dataTypeNames[1],
-            description: null,
-            is_active: true,
-            created_at: expect.any(Date),
-            updated_at: expect.any(Date),
-          },
-          {
-            id: expect.any(Number),
-            name: dataTypeNames[2],
-            description: null,
-            is_active: true,
-            created_at: expect.any(Date),
-            updated_at: expect.any(Date),
-          },
-        ]),
+        expect.arrayContaining(
+          dataTypes.map((dataType) =>
+            expect.objectContaining({
+              id: expect.any(Number),
+              name: dataType.name,
+              description: null,
+              is_active: true,
+            }),
+          ),
+        ),
+      );
+    });
+
+    it('returns all data types from cache successfully', async () => {
+      const dataTypes = [
+        { name: faker.lorem.slug() },
+        { name: faker.lorem.slug() },
+        { name: faker.lorem.slug() },
+      ];
+      const rows = await sql`
+        INSERT INTO account_data_types ${sql(dataTypes, 'name')} RETURNING (id)`;
+
+      // cache the data types
+      await target.getDataTypes();
+
+      // delete the data types from the database
+      await sql`
+        DELETE FROM account_data_types WHERE id 
+          IN ${sql(rows.map((id) => id.id))}`;
```

Comment:
As before regarding database and cache out of sync.

Reply:
Answered in https://github.com/safe-global/safe-client-gateway/pull/1760#discussion_r1679010289